lib/mail/notify/delivery_method.rb

```diff
@@ -31,12 +31,18 @@ def email_params
         {
           email_address: @mail.to.first,
           template_id: @mail[:template_id].to_s,
-          personalisation: @personalisation.to_h
+          personalisation: @personalisation.to_h,
+          email_reply_to_id: optional_param(:reply_to_id),
+          reference: optional_param(:reference)
         }
       end
 
+      def optional_param(name)
+        @mail[name].presence&.to_s
```

Comment:
Is this necessary?

Reply:
Yea, I think so. We need to call `to_s` on the field because it comes through as an object. But because it's optional sometimes it's `nil`. This should cover both cases.

Reply:
Aren't both optional params just strings?

Reply:
Not at this point it's comes in as a `Mail::Field` object if it's there.

```
#<Mail::Field 0x7fdfa7e23c70 @name="reply-to-id" @unparsed_value="custom-reply-to-id" @charset="UTF-8" @field_order_id=100>
```

Reply:
Oh, course I forgot they got parsed like that